src/main/java/com/gravity9/mongocse/MongoChangeStreamWorker.java

```diff
@@ -100,12 +100,7 @@ public void run() {
             .fullDocumentBeforeChange(mongoConfig.getFullDocumentBeforeChange())
             .maxAwaitTime(mongoConfig.getMaxAwaitTimeInMs(), MILLISECONDS);
 
-        if (resumeToken != null) {
-            log.info("Resuming change stream for partition {} on collection {} with token: {}", partition, mongoConfig.getCollectionName(), resumeToken);
-            watch.resumeAfter(buildResumeToken(resumeToken));
-        } else {
-            log.info("No resume token found for partition {} on collection {}, starting fresh", partition, mongoConfig.getCollectionName());
-        }
+        resumeIfTokenValid(watch);
```

Comment:
I thought about this but unfortunately I didn't find a way to do that. I thought about creating a replica set with 0 oplog size but:
a) it's not easily possible in Java
b) the miminum oplog size in Mongo i 990 MB - it would take a lot of time to fill it up to push the token out and as such - it's not feasible to do in a test